src/extension/sdk/dev_tools/manager.ts

```diff
@@ -376,8 +376,24 @@ export class DevToolsManager implements vs.Disposable {
 		}
 
 		if (vmServiceUri) {
-			const exposedUrl = await envUtils.exposeUrl(vmServiceUri, this.logger);
-			queryParams.uri = exposedUrl;
+			/**
+			 * In some environments (for ex. g3), the VM Service/DDS could be running on
+			 * the end user machine (eg. Mac) while the extension host is an SSH remote
+			 * (eg. Linux).
+			 *
+			 * `clientVmServiceUri` indicates a URI that is already accessible on the end
+			 * user machine without forwarding. `vmServiceUri` indicates a URI that is
+			 * accessible to the extension host.
+			 *
+			 * If a `clientVmServiceUri` exists, use it directly instead of trying to
+			 * forward a URI from the extension host.
+			 */
```

Comment:
I would put this comment (at least in part) on the `DartDebugSessionInformation` interface so it'll be visible from other code using these projects.